### PR TITLE
Feature/order summary plumbing

### LINKED ISSUE #214: Order summary plumbing
depends on #210

Order Summary component and story has already been created but no real data has been used.

This ticket is to wire up the component to real data from feathers.

related to #178

### PR DESCRIPTION
**Pull request by iainkirkpatrick**
closes #214

### CODE REVIEW

**Comment by iainkirkpatrick** [comment]
update from 10/11 from iain:

- updated some of the order getters, now there's a getter to get 'currentOrderOrderPlansByAgent', which is the kind of data shape that the `OrderSummary` component wants (render an `AgentOrderSummary` per agent)
- passing the data down from the `ViewOrderSummaryTask` container and component to the `OrderSummary` component, but not yet all the pieces required eventually by `AgentOrderSummary` - more queries, more getters probably needed in `ViewOrderSummaryTask` container

@NotThatSmith

**Comment by michael-smith-nz** [comment]
Day roundup:

- Got past the Ramda block by using ramda selectively (can refactor once I calm down 😉)
- Next step is to add the agent object to order plan via a getter. Started the scaffolding

cc: @iainkirkpatrick

**Comment by iainkirkpatrick** [comment]
update: order summary is displaying!!!

**Comment by iainkirkpatrick** [comment]
I think this ticket is ready now for review - the data has been derived and wired up correctly 🎉  (I've tested it, but probably worth someone else testing combinations of products / priceSpecs in an order).
Lots of refactoring to do at some stage, particularly in the tasks service `works.js` file.

@ahdinosaur it'd be great to get your eyes on this if you have time for some code reading + feedback! We may plow on with other tickets in the meantime though, as there is a need for some fixes / features by Monday / Wednesday next week.

NB. this also contains commits from #268

File: db/seeds/3-orders-and-intents.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    supplierAgentId: 3,
+    adminAgentId: 8
+  }
+
+  return knex('orders').insert(order).returning('id')
+    .then(([orderId]) => {
+      console.log(orderId)
+      const orderIntents = [{
+        agentId: 8,
+        desiredQuantity: 11,
```
Comment (ahdinosaur) [inline db/seeds/3-orders-and-intents.js:13]: i guess i missed this in a previous review, but all numbers should be strings, so `desiredQuantity` should be a string. not specific to this pull request, but figured worth mentioning.

File: tasks/components/ViewOrderSummaryTask.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export default (props) => {
+  const { currentOrderOrderPlansByAgent } = props
+  if (isEmpty(currentOrderOrderPlansByAgent)) return null
+  if (anyOrderPlansMissingPriceSpecs(currentOrderOrderPlansByAgent)) return null
+  if (anyOrderPlansMissingAgents(currentOrderOrderPlansByAgent)) return null
+  if (anyOrderPlansMissingProducts(currentOrderOrderPlansByAgent)) return null
+  if (anyOrderPlansMissingAgentProfiles(currentOrderOrderPlansByAgent)) return null
+  if (anyOrderPlansMissingProductResourceTypes(currentOrderOrderPlansByAgent)) return null
+  return h(OrderSummary, props)
```
Comment (ahdinosaur) [inline tasks/components/ViewOrderSummaryTask.js:21]: defensive for the win!

File: tasks/containers/ViewOrderSummaryTask.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      resourceTypes,
+      agents,
+      profiles
+    },
+    query: (props) => {
+      var queries = []
+      const { taskPlan, selected } = props
+      const { currentOrderOrderPlansByAgent } = selected
+
+      const getPriceSpecIds = pipe(
```
Comment (ahdinosaur) [inline tasks/containers/ViewOrderSummaryTask.js]: nice ramdas

File: tasks/services/works.js [hunk 4]
```diff
@@ -18,15 +18,139 @@ const hooks = {
...
+    })
+    .then((order) => {
+      return orderIntents.find({
+        query: {
+          orderId: order.id
+        }
+      })
+    })
+    .then((queriedOrderIntents) => {
+      // TODO: IK: similar logic will also be needed on the client for showing the current state of the order per product - i.e. which priceSpec is currently going to be enforced
```
Comment (ahdinosaur) [inline tasks/services/works.js]: sweet algorithm comments!

**Review by ahdinosaur — COMMENTED** [review]
woop woop

**Pull request merged**